packages/espressocash_app/lib/features/outgoing_dln_payments/screens/confirmation_screen.dart

```diff
@@ -0,0 +1,308 @@
+import 'dart:async';
+
+import 'package:dfunc/dfunc.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+import '../../../core/amount.dart';
+import '../../../core/callback.dart';
+import '../../../core/flow.dart';
+import '../../../core/presentation/format_amount.dart';
+import '../../../di.dart';
+import '../../../l10n/device_locale.dart';
+import '../../../l10n/l10n.dart';
+import '../../../ui/content_padding.dart';
+import '../../../ui/dialogs.dart';
+import '../../../ui/slider.dart';
+import '../../accounts/models/account.dart';
+import '../../blockchain/models/blockchain.dart';
+import '../models/dln_payment.dart';
+import '../models/payment_quote.dart';
+import '../services/confirm_payment_bloc.dart';
+
+class OutgoingDlnConfirmationScreen extends StatefulWidget {
+  const OutgoingDlnConfirmationScreen({
+    super.key,
+    required this.onConfirm,
+    required this.amount,
+    required this.receiverAddress,
+    required this.blockchain,
+  });
+
+  final Callback1<PaymentQuote> onConfirm;
+  final CryptoAmount amount;
+  final String receiverAddress;
+  final Blockchain blockchain;
+
+  @override
+  State<OutgoingDlnConfirmationScreen> createState() =>
+      _OutgoingDlnConfirmationScreenState();
+}
+
+class _OutgoingDlnConfirmationScreenState
+    extends State<OutgoingDlnConfirmationScreen> {
+  late final ConfirmPaymentBloc _bloc;
+  Timer? _timer;
+
+  @override
+  void initState() {
+    super.initState();
+
+    _bloc = sl<ConfirmPaymentBloc>(
+      param1: DlnPayment(
+        inputAmount: widget.amount,
```

Comment:
Instead of passing it as parameter, create another event like `init` and pass it there. `factoryParam` should be avoided as much as possible, they are not compile-time safe.